reV/supply_curve/tech_mapping.py

```diff
@@ -393,105 +351,134 @@ def _check_fout(self):
+                                self._excl_row_slices,
+                                self._excl_col_slices,
+                            )
+                            n_rows = row_slice.stop - row_slice.start
+                            n_cols = col_slice.stop - col_slice.start
+                            result_shape = (n_rows, n_cols)
+                            indices[row_slice, col_slice] = result[j].reshape(
+                                result_shape
+                            )
                 n_finished += 1
                 logger.info(
-                    "Parallel TechMapping futures collected: "
-                    "{} out of {}".format(n_finished, len(futures))
+                    "Parallel TechMapping batches completed: "
+                    f"{n_finished} out of {n_batches}"
                 )
 
-                i = futures[future]
-                result = future.result()
-
-                for j, gid in enumerate(gid_chunks[i]):
-                    row_slice, col_slice = self._get_excl_slices(
-                        gid,
-                        self._sc_row_indices,
-                        self._sc_col_indices,
-                        self._excl_row_slices,
-                        self._excl_col_slices,
-                    )
-                    ind_slice = iarr[row_slice, col_slice].flatten()
-                    indices[ind_slice] = result[j]
-
-        indices = indices.reshape(self._excl_shape)
-
-        return indices
-
     @classmethod
     def run(
         cls,
         excl_fpath,
         res_fpath,
-        dset=None,
-        sc_resolution=2560,
+        dset,
+        sc_resolution=64,
         dist_margin=1.05,
         max_workers=None,
         points_per_worker=10,
+        batch_size=100,
     ):
         """Run parallel mapping and save to h5 file.
 
         Parameters
         ----------
         excl_fpath : str
-            Filepath to exclusions h5 (tech layer). dset will be
-            created in excl_fpath.
+            Filepath to exclusions data HDF5 file. This file must must contain
+            latitude and longitude datasets.
         res_fpath : str
-            Filepath to .h5 resource file that we're mapping to.
-        dset : str, optional
-            Dataset name in excl_fpath to save mapping results to, if None
-            do not save tech_map to excl_fpath, by default None
+            Filepath to HDF5 resource file (e.g. WTK or NSRDB) to which
+            the exclusions will be mapped. Can refer to a single file (e.g.,
+            "/path/to/nsrdb_2024.h5" or a wild-card e.g.,
+            "/path/to/nsrdb_{}.h5")
+        dset : str
+            Dataset name in the `excl_fpath` file to which the the
+            techmap (exclusions-to-resource mapping data) will be saved.
+
+            .. Important:: If this dataset already exists in the h5 file,
+              it will be overwritten.
         sc_resolution : int | None, optional
```

Comment:
I think for proper sphinx formatting, we'll need a newline here
```suggestion
            .. Important:: If this dataset already exists in the h5 file,
               it will be overwritten.

        sc_resolution : int | None, optional
```